packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -434,8 +437,12 @@ class _CircularProgressIndicatorPainter extends CustomPainter {
       canvas.drawArc(Offset.zero & size, 0, _sweep, false, backgroundPaint);
     }
 
-    if (value == null) { // Indeterminate
+    if (value == null && strokeCap == null) {
+      // Indeterminate
       paint.strokeCap = StrokeCap.square;
+    } else {
+      // Butt when determinate (value != null) && strokeCap == null;
+      paint.strokeCap = strokeCap ?? StrokeCap.butt;
```

Comment:
`StrokeCap.butt` was not here before. Why this change?

Reply:
it wasn't set because it was the default value, we must explicit set it now